src/routes/+layout.server.ts

```diff
@@ -2,9 +2,12 @@ import { loadPageMetadataMap } from "./loadPageMetadataMap";
 import { loadSiteTitle } from "$lib/components/Header/Title/Title.server";
 import { loadMainNav, loadSecondaryNav } from "$lib/components/Header/Nav/Nav.server";
 
-export const load = async () => ({
-  pageMetadataMap: loadPageMetadataMap(),
-  siteTitle: loadSiteTitle(),
-  secondaryNavItems: loadSecondaryNav(),
-  navItems: await loadMainNav(),
-});
+export const load = async () => {
+  const pageMetadataMap = await loadPageMetadataMap();
+  return {
+    pageMetadataMap,
+    siteTitle: loadSiteTitle(),
+    navItems: loadMainNav(pageMetadataMap),
```

Comment:
this is less expensive than the `loadPageMetadataMap` query/transformation, but as a follow-up in the same vein as comments on #273, should this get cached and the value for `navItems` stored in a cookie, as well?

Reply:
Yeah, I think that ideally all of these data structures (the page metadata map, the main menu items, and the sidenavs map) should be cached in some way. They could easily get out-of-sync with each other if we don't update them all at once.